include/multipass/cli/formatted_output.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2018 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef MULTIPASS_FORMATTED_OUTPUT_H
+#define MULTIPASS_FORMATTED_OUTPUT_H
+
+#include <multipass/rpc/multipass.grpc.pb.h>
+
+#include <string>
+
+namespace multipass
+{
+class FormattedOutput
```

Comment:
I tend to avoid -er names but given the interface methods below, this seems more like a "Formatter" than something holding formatted output.

With a name like FormattedOutput I would expect something like

```
class FormattedOutput
{
public:
    virtual void write(const InfoReply& reply) = 0;
    virtual void write(const ListReply& reply) = 0;
    virtual std::string str() const = 0;
};
```
Then with an operator << overload:
```
std::ostream& operator<<(std::ostream& out, const mp::FormattedOutput& f)
{
    out << f.str();
    return out;
}
```